extensions/amp-selector/0.1/amp-selector.js

```diff
@@ -220,6 +284,71 @@ export class AmpSelector extends AMP.BaseElement {
   }
 
   /**
+   * Handles click events for the selectables.
+   * @param {!Event} event
+   */
+  clickHandler_(event) {
+    let el = dev().assertElement(event.target);
+    if (!el) {
+      return;
+    }
+    if (!el.hasAttribute('option')) {
+      el = closest(el, e => e.hasAttribute('option'), this.element);
+    }
+    if (el) {
+      this.onOptionSelected_(el);
+    }
+  }
+
+  /**
+   * Handles keyboard events for the selectables.
+   * @param {!Event} event
+   */
+  keyDownHandler_(event) {
+    // Make currently selected option unfocusable
+    this.options_[this.focusedIndex_].tabIndex = -1;
+
+    const dir = this.win.document.body.getAttribute('dir') || 'ltr';
+    const ltr = dir != 'rtl';
+    let delta = 0;
+
+    switch (event.keyCode) {
+      case 37: // Left
+        delta = ltr ? -1 : 1;
+        break;
+      case 38: // Up
+        delta = -1;
+        break;
+      case 39: // Right
+        delta = ltr ? 1 : -1;
+        break;
+      case 40: // Down
+        delta = 1;
+        break;
+    }
+
+    if (delta == 0) {
+      return;
+    }
+    event.preventDefault();
+
+    this.focusedIndex_ = (this.focusedIndex_ + delta) % this.options_.length;
```

Comment:
1. Done
2. Will expand on comments in the constructor.
3. I think that may end up being more difficult to read. The `updateFocus_` method introduced in this PR is almost what  you described, but it doesn't assume that the amp-selector has focus. If I were to move the logic from here to there, that method would become a lot more difficult to read. I could refactor some of this out into its own method but it would only be used here. Let me know if you see a way to simplify this as I agree that a single method to manage focus would be nice.